src/metric/FARCompoundMetric.py

```diff
@@ -0,0 +1,373 @@
+
+        Args:
+            cmm (CompoundMetricManager): Manager used by metric. Population and dataset is derived
+                from this manager.
+            smm (SampleMetricManager, optional): Manager used by metric.
+        """
+        super(FARCompoundMetric, self).__init__(FAR_NAME, cmm, smm)
+
+        # Define threshold linspace
+        self._thresholds = self._get_threshold_space()
+
+        # Init storage structure for this metric
+        self._file_jar = get_file_jar()
+        self._far_ds_vs_ds = None
+        self._far = None
+        self._filter_name_dict = dict()
+
+    def reg_setup_modes(self) -> dict[str, SetupMode]:
+        ds_setup_modes = None
+        pop_setup_modes = None
+        for sm in self._smm.get_metric_instances():
+            if isinstance(sm, DatasetSimilaritySampleMetric):
+                ds_setup_modes = sm.reg_setup_modes()
+            elif isinstance(sm, PopulationSimilaritySampleMetric):
+                pop_setup_modes = sm.reg_setup_modes()
+        assert ds_setup_modes is not None
+        assert pop_setup_modes is not None
+
+        # Merge sample metric setup modes
+        sample_metric_setup_modes = pop_setup_modes | ds_setup_modes
+
+        # Define setup mode exclusive for FAR
+        ds_vs_ds_setup_mode = {
+            DS_VS_DS_SETUP_NAME: SetupMode(
+                lambda _: self._setup_dataset_vs_dataset(),
+                self._is_dataset_vs_dataset_ready,
+                required_modes=[setup_name for setup_name in ds_setup_modes.keys()],
+            )
+        }
+
+        # Merge all
+        return sample_metric_setup_modes | ds_vs_ds_setup_mode
+
+    def _setup_dataset_vs_dataset(self):
+        # Calculate the dataset vs dataset similarity
+        sim = DatasetSimilaritySampleMetric.dataset_vs_dataset_similarity(self._dataset)
+
+        # Store result on disk
+        self._file_jar.store_file(
+            DS_VS_DS_FILE_NAME(self._dataset.get_name(self._dataset.get_resolution())),
+            lambda p: np.save(p, sim),
+        )
+
+    def _is_dataset_vs_dataset_ready(self):
+        return self._file_jar.has_file(
+            DS_VS_DS_FILE_NAME(self._dataset.get_name(self._dataset.get_resolution()))
+        )
+
+    def _calc_far(self, similarities: np.ndarray) -> np.ndarray:
+        #! Assumes all samples are different identities
+
+        # If a similarity is above the threshold, then this sample is considered the same
+        # identity with its neighbor, by assumption samples are not same identities.
+        # Thus, it is misclassified when they are the same identity.
+
+        # Calculates the FAR as a function of thresholds
+        return np.mean(
+            similarities.reshape(similarities.size, 1)
+            > self._thresholds.reshape(1, self._thresholds.shape[0]),
+            0,
+        )
+
+    def _get_threshold_space(self) -> np.ndarray:
+        # TODO COOL WAY TO INFERR MAX/MIN?
```

Comment:
TODO? 

If you want to keep it (completely valid), put `# TODO: ...` instead (add colon)